homework-07/script.js

```diff
@@ -0,0 +1,77 @@
+const YEAR_NOW = (new Date()).getFullYear();
+
+const user = {
+    year: prompt('Ваш рік народження:'),
+    city: prompt('Місто, в якому проживаєте:'),
+    sport: prompt('Улюблений вид спорту:')
+};
+
+let fillAge, fillCity, fillSport;
+
+if(user.year === null) {
+    alert('Шкода, що Ви не захотіли ввести свій рік народження');
+} else if(isNaN(+user.year) || +user.year <= 1900 || +user.year > YEAR_NOW) {
+    alert('Дата народження вказана невірно');
+} else {
+    user.age = YEAR_NOW - user.year;
+    fillAge = true;
+}
+
+if(user.city === null) {
+    alert('Шкода, що Ви не захотіли ввести своє місто проживання');
+} else {
+    fillCity = true;
+}
+
+if(user.sport === null) {
+    alert('Шкода, що Ви не захотіли ввести свій улюблений вид спорту');
+} else {
+    fillSport = true;
+}
+
+if(fillAge === true && fillCity === true && fillSport === true) {
```

Comment:
Замість створення змінних як `fillAge` можна було б перевіряти чи визначені властивості `age` та інші, наприклад

```suggestion
if(user.age && user.city && user.sport) {
```

Так, тут відбудеться неявне перетворення у `boolean` але усі falsy значення нас теж не влаштовіють.

Reply:
Оо, клас, дякую! Не прийшло в голову. Дякую за влучні коментарі, візьму на замітку.